### PR TITLE
[Video] Warn and return all frames when num_frames exceeds total_num_frames

### LINKED ISSUE #48073: load_video with num_frames > total_num_frames silently returns the same frame repeated
## System Info

- transformers main (commit 8c62d6f)
- Python 3.13, numpy

## Who can help?

@molbap @guarin

## Reproduction

Was sampling a fixed number of frames from short videos and noticed that when `num_frames` exceeds the actual frame count, the returned frames are all identical with no warning. Traced it to `default_sample_indices_fn`:

```python
import numpy as np
total_num_frames = 10
num_frames = 20
indices = np.arange(0, total_num_frames, total_num_frames / num_frames, dtype=int)
print(indices.tolist())
# [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
```

The step `total/num = 0.5` gets truncated to `0` by `dtype=int`, so every index is 0. The user asks for 20 frames from a 10-frame video and silently gets frame 0 repeated 20 times. No error, no warning.

This is reachable via `load_video(path, num_frames=20)` on a 10-frame video — `num_frames` is forwarded directly to `default_sample_indices_fn` with no guard. Interestingly the `fps` branch of the same function *does* check `if num_frames > total_num_frames: raise ValueError`, but the direct `num_frames` path doesn't.

## Expected behavior

At minimum a warning (or error) when `num_frames > total_num_frames`, since silently returning 20 copies of the same frame is almost never what the user wants.

Note: `get_uniform_frame_indices` (the standalone helper) uses the same `np.arange` pattern but with `.astype(int)` instead of `dtype=int`, and produces different indices for many `(total, num_frames)` combinations — the two functions are inconsistent.

I've already prepared a fix and will open a PR shortly.

### PR DESCRIPTION
**Pull request by carlszk**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48074)

## What does this PR do?
Fixes #48073

## Problem
`default_sample_indices_fn` in `src/transformers/video_utils.py` silently returns `num_frames` copies of index 0 when `num_frames > total_num_frames`. The step `total/num < 1` is truncated to `0` by `dtype=int`, so `np.arange(0, total, 0, dtype=int)` yields `[0, 0, ..., 0]`. Reachable via `load_video(path, num_frames=N)` on a video with fewer than `N` frames — the user asks for N distinct frames and silently gets one frame repeated N times. The `fps` branch already guards `if num_frames > total: raise ValueError`, but the direct `num_frames` path does not.

## Fix
When `num_frames > total_num_frames`, emit a warning and return all available frames instead of silently duplicating frame 0. Non-breaking: no exception, just a warning + sensible fallback.

## Tests
- `num_frames > total`: 10 unique frames (not 20 copies of 0) + warning emitted
- `num_frames == total`: boundary case
- `num_frames < total`: no regression

## Checklist
- [x] My PR addresses the issue
- [x] I have added tests

### CODE REVIEW

**Comment by carlszk** [comment]
Hi @molbap @guarin, CI is green (0 failures after switching to `linspace` to preserve the `num_frames` count when it exceeds `total_num_frames`). Ready for review when you have a moment! Fixes #48073.

File: src/transformers/video_utils.py [hunk 1]
```diff
@@ -332,7 +332,15 @@ def default_sample_indices_fn(metadata: VideoMetadata, num_frames=None, fps=None
...
 
     if num_frames is not None:
-        indices = np.arange(0, total_num_frames, total_num_frames / num_frames, dtype=int)
+        if num_frames > total_num_frames:
+            logger.warning_once(
+                f"`num_frames={num_frames}` exceeds the total number of frames in the video "
+                f"({total_num_frames}). Sampling {num_frames} frames uniformly (with repetition) "
+                f"from the available {total_num_frames} frames."
+            )
+            indices = np.linspace(0, max(total_num_frames - 1, 0), num_frames, dtype=int)
```
Comment (zucchini-nlp) [inline src/transformers/video_utils.py]: better raise an error imo and ask to sample less frames, see L329-332. Let's get that error, and move it around so it is raised when we have final `num_frames`

Comment (zucchini-nlp) [reply src/transformers/video_utils.py]: example - https://github.com/huggingface/transformers/blob/e12c79c582bc742029e5f12b538ec71391ace8b3/src/transformers/video_processing_utils.py#L168-L180

File: tests/utils/test_video_utils.py [hunk 2]
```diff
@@ -387,3 +388,27 @@ def test_load_video_fps(self):
...
+        # returning `num_frames` copies of index 0 (the previous behaviour due to
+        # `dtype=int` truncating the step to 0).
+        metadata = VideoMetadata(total_num_frames=10, fps=30.0)
+        with self.assertLogs("transformers.video_utils", level="WARNING") as cm:
+            indices = default_sample_indices_fn(metadata, num_frames=20)
+        self.assertEqual(len(indices), 20)
+        self.assertTrue((indices >= 0).all() and (indices < 10).all())
+        self.assertGreater(len(set(indices.tolist())), 1)  # not all the same index
+        self.assertTrue(any("exceeds the total number of frames" in msg for msg in cm.output))
+
```
Comment (zucchini-nlp) [inline tests/utils/test_video_utils.py]: can we just call `load_video` please with `num_frames` for the test and a single test will be enough. 

(actually we need to deprecate the `sample_indices_fn` since the video processors were refactored long ago, not on you though, just a sidenote for myself)

**Review by zucchini-nlp — APPROVED** [review]
